swarm/storage/mru/lookup.go

```diff
@@ -17,96 +17,94 @@
 // useful to build query strings
-func (lp *LookupParams) ToValues(values url.Values) {
-	if lp.Limit != 0 {
-		values.Set("limit", fmt.Sprintf("%d", lp.Version))
+func (lp *LookupParams) ToValues(values Values) {
+	if lp.TimeLimit != 0 {
+		values.Set("time", fmt.Sprintf("%d", lp.TimeLimit))
 	}
-	lp.UpdateLookup.ToValues(values)
+	if lp.Hint.Level != 0 {
+		values.Set("hint.level", fmt.Sprintf("%d", lp.Hint.Level))
+	}
+	if lp.Hint.Time != 0 {
+		values.Set("hint.time", fmt.Sprintf("%d", lp.Hint.Time))
+	}
+	lp.View.ToValues(values)
 }
 
-// NewLookupParams constructs a LookupParams structure with the provided lookup parameters
-// if period = 0 and version = 0, the last resource version will be looked up
-// if period !=0 and version = 0, the last version in that period will be looked up
-// if both period, version !=0, that specific period, version will be looked up
-func NewLookupParams(view *View, period, version uint32, limit uint32) *LookupParams {
+// LookupBefore constructs an UpdateLookup structure to find updates on or before `time`
+// if time == 0, the latest update will be looked up
+func LookupBefore(view *View, time uint64, hint lookup.Epoch) *LookupParams {
 	return &LookupParams{
-		UpdateLookup: UpdateLookup{
-			Period:  period,
-			Version: version,
-			View:    *view,
-		},
-		Limit: limit,
+		TimeLimit: time,
+		View:      *view,
+		Hint:      hint,
 	}
 }
 
 // LookupLatest generates lookup parameters that look for the latest version of a resource
-func LookupLatest(view *View) *LookupParams {
-	return NewLookupParams(view, 0, 0, 0)
-}
-
-// LookupLatestVersionInPeriod generates lookup parameters that look for the latest version of a resource in a given period
-func LookupLatestVersionInPeriod(view *View, period uint32) *LookupParams {
-	return NewLookupParams(view, period, 0, 0)
+func LookupLatest(view *View, hint lookup.Epoch) *LookupParams {
+	return LookupBefore(view, 0, hint)
 }
 
-// LookupVersion generates lookup parameters that look for a specific version of a resource
-func LookupVersion(view *View, period, version uint32) *LookupParams {
-	return NewLookupParams(view, period, version, 0)
-}
-
-// UpdateLookup represents the components of a resource update search key
+// UpdateLookup represents the components of a resource update search key.
+// it is also used to specify constraints when performing an update lookup
 type UpdateLookup struct {
-	View
-	Period  uint32
-	Version uint32
+	View         `json:"view"`
+	lookup.Epoch `json:"epoch"`
 }
 
 // UpdateLookup layout:
 // ResourceIDLength bytes
 // userAddr common.AddressLength bytes
 // 4 bytes period
 // 4 bytes version
-const updateLookupLength = viewLength + 4 + 4
+const updateLookupLength = viewLength + lookup.EpochLength
 
 // UpdateAddr calculates the resource update chunk address corresponding to this lookup key
 func (u *UpdateLookup) UpdateAddr() (updateAddr storage.Address) {
```

Comment:
call this func `Addr`

Reply:
We discussed this in previous PRs. We left it as `UpdateAddr` because this struct gets embedded in others and it could create confusion when methods merge.